src/clients/c++/request.h

```diff
@@ -347,6 +354,13 @@ class InferContext {
     /// model configuration. Variable-size dimensions are reported as
     /// -1.
     virtual const DimsList& Dims() const = 0;
+
+    // Set the shard memory object for the batch tensor output
```

Comment:
Same comments as for input. Till now Output has only returned information about the model configuration so using it to set something may take some work in the implementation.